artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -132,13 +132,14 @@ impl Integer {
             // ```
             #[allow(clippy::unnested_or_patterns)]
             match u8::try_from(self.as_i64()) {
-                // ASCII encoding | Binary/ASCII-8BIT encoding
                 // Without `Encoding` support, these two arms are the same
-                Ok(chr @ 0..=127) | Ok(chr @ 128..=255) => {
-                    // Create a single byte `String` from the character given by
-                    // `self`.
-                    Ok(vec![chr])
-                }
+
+                // ASCII encoding - chr[0 - 127]
+                // Binary/ASCII-8BIT encoding - chr[128 - 255]
+
+                // Create a single byte `String` from the character given by `self`.
+                Ok(chr @ 0..=127) => Ok(spinoso_string::String::ascii(vec![chr])),
+                Ok(chr @ 128..=255) => Ok(spinoso_string::String::binary(vec![chr])),
                 _ => {
```

Comment:
I think this could be converted into an `Err(_)` pattern which would ensure we're exhaustively matching the `Ok` variants above.

```suggestion
                Err(_) => {
```